plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/HeadSampler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.sampler;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+import java.util.List;
+
+/**
+ * HeadBased sampler
+ */
+public class HeadSampler implements Sampler {
+    private final Sampler defaultSampler;
+
+    /**
+     * Creates Head based sampler
+     * @param defaultSampler defaultSampler
+     */
+    public HeadSampler(Sampler defaultSampler) {
+        this.defaultSampler = defaultSampler;
+    }
+
+    @Override
+    public SamplingResult shouldSample(
+        Context parentContext,
+        String traceId,
+        String name,
+        SpanKind spanKind,
+        Attributes attributes,
+        List<LinkData> parentLinks) {
+
+        if (attributes.get(AttributeKey.stringKey("trace")) != null && Boolean.parseBoolean(attributes.get(AttributeKey.stringKey("trace"))) ) {
```

Comment:
Make it a constant, Later on we will replace it with https://github.com/opensearch-project/OpenSearch/pull/9415/files#diff-f7fd27ef940cf0f44f7e5db222621c0ec2ff93ad3ccf857df681e7e24679ee44R44

Reply:
Also, we need to see how can we bring this logic out which is specific to server not otel.